File: 3107-minimum-operations-to-make-median-of-array-equal-to-k/3107-minimum-operations-to-make-median-of-array-equal-to-k.cs
public class Solution {
    public long MinOperationsToMakeMedianK(int[] nums, int k) {
        long res = 0;
        Array.Sort(nums);
        int numsSize = nums.Length;
        for (int i = 0; i <= numsSize / 2; i++)
        {
            res += Math.Max(0, nums[i] - k); // decrease excess numbers
        }

        for (int i = numsSize / 2; i < numsSize; i++)
        {
            res += Math.Max(0, k - nums[i]);
        }

        return res;
    }
}

File: 3107-minimum-operations-to-make-median-of-array-equal-to-k/3107-minimum-operations-to-make-median-of-array-equal-to-k.c
int compare(const void *a, const void *b) {
    return (*(int *)a - *(int *)b);
}

long long minOperationsToMakeMedianK(int* nums, int numsSize, int k) {
    long long res = 0;
    qsort(nums, numsSize, sizeof(int), compare);
    for (int i = 0; i <= numsSize/2; i++){
        res += fmax(0, nums[i] - k); // decrease excess numbers
    }
    for (int i = numsSize/2; i < numsSize; i++){
        res += fmax(0,k - nums[i]);
    }
    return res;    
}

//2,5,5,6,8 k= 7